### PR TITLE
feat(repository-json-schema): enumerate fields

### PR DESCRIPTION
**Pull request by ghsamm**
enumerate properties of the Entity in the schema of the fields filter

## Checklist

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

### CODE REVIEW

File: packages/repository-json-schema/src/filter-json-schema.ts [hunk 1]
```diff
@@ -95,3 +90,24 @@ export function getWhereJsonSchemaFor(modelCtor: typeof Model): JsonSchema {
...
+export function getFieldsJsonSchemaFor(modelCtor: typeof Model): JsonSchema {
+  const schema: JsonSchema = {
+    type: 'object',
+    properties: Object.assign(
+      {},
+      ...Object.keys(modelCtor.definition.properties).map(k => ({
+        [k]: {type: 'boolean'},
+      })),
+    ),
+    additionalProperties: true,
```
Comment (bajtos) [inline packages/repository-json-schema/src/filter-json-schema.ts]: I think `additionalProperties` should be set to:
- `false` when the model is in strict mode and does not allow any additional properties (this is the default in LB4)
- `true` when the model disabled strict mode and allows any additional properties (FWIW, this is the default in LB3)

Would you mind changing your pull request accordingly?

I think you can find the relevant model setting via `modelCtor.definition.settings.strict`.

Comment (ghsamm) [reply packages/repository-json-schema/src/filter-json-schema.ts]: done

**Comment by raymondfeng** [comment]
Please rebase your branch to the latest master instead of using merge.
```sh
git remote add upstream git@github.com:strongloop/loopback-next.git
git pull --rebase upstream master
git push -f
```

**Comment by ghsamm** [comment]
@raymondfeng I had to rebase on branch context-watcher for tests to work.

What I understood is there is new config for the CI, right ?

**Comment by raymondfeng** [comment]
@ghsamm You should NOT rebase to `context-watcher` branch as it's still WIP. Your PR branch should be rebased to `master` only as it's the target for merge.

**Comment by raymondfeng** [comment]
@ghsamm Here is what I have to do to rebase your PR against the latest master.

```sh
git rebase -i HEAD~16 (this opens an editor, you should remove the commits from `context-watcher`)
git remote add upstream git@github.com:strongloop/loopback-next.git
git pull --rebase upstream master
git push -f
```

**Comment by ghsamm** [comment]
> @ghsamm You should NOT rebase to `context-watcher` branch as it's still WIP. Your PR branch should be rebased to `master` only as it's the target for merge.

Yes I know, I had a weird bug in travis CI so I had to re-rebase the commit for the build to restart. It worked the second time although it is the same commit !!

**Comment by bajtos** [comment]
Landed, thank you for the contribution!

**Pull request merged**